src/PoolManager.sol

```diff
@@ -291,6 +297,13 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, ERC6909Claims {
         _burnFrom(from, id, amount);
     }
 
+    // Allows one caller to resolve a debt on behalf of another address
+    function payOnBehalf(Currency currency, address caller, uint256 amount) external isLocked {
```

Comment:
with this function can't the user's `accountDelta` and `balanceDelta` return from a function be out of whack?? seems unideal